decoder.go

```diff
@@ -16,9 +16,60 @@ const (
 	defaultMaxSize = 16000
 )
 
+type option[T Decoder | Encoder] func(d *T)
+
 // NewDecoder returns a new Decoder.
-func NewDecoder() *Decoder {
-	return &Decoder{cache: newCache(), maxSize: defaultMaxSize}
+func NewDecoder(opts ...option[Decoder]) *Decoder {
+	d := &Decoder{cache: newCache(), maxSize: defaultMaxSize}
+	for _, opt := range opts {
+		opt(d)
+	}
+	return d
+}
+
+// WithAliasTagDecoderOpt changes the tag used to locate custom field aliases.
+// The default tag is "schema".
+func WithAliasTagDecoderOpt(tag string) option[Decoder] {
+	return func(d *Decoder) {
+		d.SetAliasTag(tag)
+	}
+}
+
+// WithZeroEmptyDecoderOpt controls the behaviour when the decoder encounters empty values.
```

Comment:
I see that the comments are duplicates of the original "setter" function. Unless we are planning to remove those functions and relying on options, I think it's enough to point to the original function, e.g:
```
// Option to set zeroEmpty on a new Decoder. See ZeroEmpty for more information. 
```

Reply:
> I see that the comments are duplicates of the original "setter" function. Unless we are planning to remove those functions and relying on options, I think it's enough to point to the original function, e.g:
> 
> ```
> // Option to set zeroEmpty on a new Decoder. See ZeroEmpty for more information. 
> ```

That can be a bit confusing `zeroEmpty` is an internal struct field so it's not visible in godoc. So referring to it maybe isn't the best idea. What about just the "See" part is enough? With a link:

```go
// See [Decoder.ZeroEmpty] for more information.
```